github.com.conf.template

```diff
@@ -0,0 +1,31 @@
+# Proxy git's smart http protocol to only certain organisations on github.com.
+#
+# https://git-scm.com/docs/http-protocol
+#
+# The git http protocol design makes it non-trivial to distinguish between push
+# and pull - they are identical in terms of http verb and path.
+# 
+# We are keen to block push, but in practice we still need so support it for
+# now. We may be able to limit it by restricting the client_max_body_size to
+# very small size in future.
+server {
+
+    server_name github-proxy.opensafely.org;
+    root /var/www/html/;
+    listen ${PORT};
+
+    # The default nginx request body size is 1M, we make it explicit and bump
+    # to support current use cases
+    client_max_body_size 2M;
+
+    # only opensafely and opensafley-core orgs can be accessed
+    location ~ /(opensafely|opensafely-core)/.*/(info|git-upload-pack|git-receive-pack) {
```

Comment:
The regex here doesn't seem to be consistent with the use of `git-upload-pack` etc in the query fragment -- it looks like they're part of the path here, but that's not right, is it?

Reply:
It is right! At least, if I've understood it correctly. It certainly seems to work. 

tl;dr: it works by appending `/somepath` to the end of the base repo url. So depending on your base repo url, it could be an actual path component (like github does) or part of a query string.

Basically, git's http protocol is not very http like. Search for $GIT_URL here for examples: https://git-scm.com/docs/http-protocol